### ISSUE TITLE
Nested .editorconfig not supported

### ISSUE THREAD

**Issue by platinumazure** [platinumazure/eslint-plugin-editorconfig#1]
This plugin could probably work with a single `.editorconfig` in the root directory of a project. However, nested `.editorconfig` files will not be picked up, and in fact the plugin's exported configuration will not even be executed more than once.

Unfortunately this is blocked by ESLint, since it is not designed to re-process configuration multiple times (usually for good reason, since performance could tank). We will need to reach out to ESLint to see if there is a way we could change this on an opt-in basis for specific plugins or configurations.

**Comment by platinumazure** [comment]
Tracking the upstream issue at eslint/eslint#8943.

**Comment by platinumazure** [comment]
Update: Since .editorconfig uses glob patterns and since ESLint supports glob-based configuration, our best bet may be to read .editorconfig files in CWD + descendant directories, and combine the glob patterns into a single configuration. Hoping to explore that in the next couple of weeks.

**Comment by platinumazure** [comment]
Apparently I solved this (or at least partially solved this) almost three years ago.

Wow.

I guess I'll close this for now. If anyone has trouble with nested .editorconfig files, please open a new issue and describe your use case in detail. Thanks!
